PORTAL/jobs/_pyperformance.py

```diff
@@ -1583,13 +1583,13 @@ def copy_to(
     ) -> "PyperfResults":
         if self._resfile is None:
             raise ValueError
-        if os.path.exists(self._resfile.filename):
-            resfile = self._resfile.copy_to(filename, resultsroot,
-                                            compressed=compressed)
+        resfile: PyperfResultsFile
+        if isinstance(filename, PyperfResultsFile):
+            resfile = filename
```

Comment:
`PyperfResultsFile.copy_to` seems to have an optimization where it just copies the original file (using `shutil.copyfile`) without regard to the fact that the data has changed.  That seems like the source of all files containing all of the results.  I think it's safer to just write out the JSON from the data every time so we are ensured we get exactly what's in the object.  So I did that here and removed `PyperfResultsFile.copy_to` below to avoid confusion in the future.

Reply:
Yeah, we'd need a (private) flag indicating that `self.data` does not match `self._resfile` any more.  That would be set in `split_benchmarks()`.  I'll leave comments to demonstrate.

If you think that's too much then I'm fine with always calling `resfile.write().

Reply:
Good catch, BTW.